slash/resuming.py

```diff
@@ -148,19 +156,20 @@ def get_tests_from_previous_session(session_id, get_successful_tests=False):
                     other_tests.append(new_entry)
             return first_tests + other_tests
     _logger.debug('No local entry for session {}, searching remote session', session_id)
-    return get_tests_from_remote_session(session_id, get_successful=True)
+    return get_tests_from_remote_session(session_id, get_successful=get_successful_tests)
 
 
 def get_tests_from_remote_session(session_id, get_successful=False):
     unstarted_first, failed_first = config.root.resume.unstarted_first, config.root.resume.failed_first
+    unstarted_only, failed_only = config.root.resume.unstarted_only, config.root.resume.failed_only
     backslash_plugin = manager.get_active_plugins().get('backslash', None)
     if not backslash_plugin:
         raise CannotResume("Could not find backslash plugin")
     if not hasattr(backslash_plugin, 'is_session_exist') or not hasattr(backslash_plugin, 'get_tests_to_resume'):
         raise CannotResume("Backslash plugin doesn't support remote resuming")
     if not backslash_plugin.is_session_exist(session_id):
         raise CannotResume("Could not find resume data for session {}".format(session_id))
-    remote_tests = backslash_plugin.get_tests_to_resume(session_id, get_successful, unstarted_first, failed_first)
+    remote_tests = backslash_plugin.get_tests_to_resume(session_id, get_successful, unstarted_first, failed_first, unstarted_only, failed_only)
```

Comment:
This should depend on the version of backslash-python we have installed to avoid breaking existing environments